dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/xmlworkflow/actions/GeneratePDFAction.java

```diff
@@ -0,0 +1,122 @@
+
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.dspace.uclouvain.core.utils.MetadataUtils;
+import org.dspace.uclouvain.pdfAttestationGenerator.factory.PDFAttestationGeneratorFactory;
+import org.dspace.uclouvain.pdfAttestationGenerator.handlers.PDFAttestationGeneratorHandler;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Email;
+import org.dspace.eperson.EPerson;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.xmlworkflow.state.Step;
+import org.dspace.xmlworkflow.state.actions.ActionResult;
+import org.dspace.xmlworkflow.state.actions.processingaction.ProcessingAction;
+import org.dspace.xmlworkflow.storedcomponents.XmlWorkflowItem;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+* Main action to generate a PDF attestation for the workflow item if his type is handled
+*/
+public class GeneratePDFAction extends ProcessingAction {
+    
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    MetadataUtils metadataUtils;
+
+    private String source = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("dspace.dir");
+    
+    @Override
+    public void activate(Context c, XmlWorkflowItem wf){}
+
+    /**
+    * Create an email with some information from the submission, attach the PDF attestation and send it to the submitter
+    */
+    @Override
+    public ActionResult execute(Context c, XmlWorkflowItem wfi, Step step, HttpServletRequest request){
+        // UUID of the current workflow item
+        UUID uuid = wfi.getItem().getID();
+        try {
+            // Recover the correct handler for this submission
+            PDFAttestationGeneratorHandler handler = PDFAttestationGeneratorFactory.getInstance().getHandlerInstance(uuid);
+            // If the type of the submission is supported, we have an handler
+            if (handler != null) {
+
+                Item dspaceItem = itemService.find(c, uuid);
+                EPerson submitter = dspaceItem.getSubmitter();
+                HashMap<String, List<String>> map = metadataUtils.getValuesHashMap(dspaceItem.getMetadata());
+
+                try {
+                    String templatePath = this.source + "/config/emails/pdf_attestation";
+                    InputStream pdfAttestation = handler.getAttestationAsInputStream(uuid);
+
+                    // Generate email
+                    Email email = Email.getEmail(templatePath);
+                    // TODO: Use submitter's email
+                    // email.addRecipient(submitter.getEmail());
+                    email.addRecipient("michael.pourbaix@uclouvain.be");
```

Comment:
For next PR : 
* recipients could be defined by a methods. This methods could be defined as param in Dspace configuration.
* possible to manage "TO", "CC" and maybe "BCC" recipient types